packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3920,6 +3920,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       : null;
   }
 
+  TextPosition _moveToParagraphBoundary(TextPosition extent, bool forward, TextBoundary textBoundary) {
```

Comment:
can you just use _moveBeyondTextBoundary instead?

Reply:
That worked! I think I was tying myself up here when trying to implement `getLeadingTextBoundaryAt` and `getTrailingTextBoundaryAt`.